### ISSUE TITLE
Error while using Pyvista with vtk-osmesa

### ISSUE THREAD

**Issue by SMoraisAnsys** [pyvista/pyvista#6234]
### Describe the bug, what's wrong, and what you expected.

While adding python3.12 compatibility in [pyaedt](https://github.com/ansys/pyaedt), some of our tests involving `pyvista` started to fail. From what we've been able to understand, the bug appears when we load a mesh and should be related to #5650.

Following comments on https://github.com/pyvista/pyvista/issues/6200, we tried to use the latest version of `pyvista` with :
1. vtk-osmesa==9.3.0 from https://gitlab.kitware.com/api/v4/projects/13/packages/pypi/simple (like in your docs action)
2. vtk-osmesa==9.3.20231014.dev0 from https://wheels.vtk.org (oldest dev found)
3. vtk-osmesa==9.3.20240525.dev0 from https://wheels.vtk.org (newest dev found)

Unfortunately all of this approaches failed.

A work around is to work with `pyvista==0.42.3` and, for example, `vtk-osmesa==9.3.20231014` succeeded.
However, that means that `pyvista` versions flagged as compatible with python3.12 (aka 0.43.Z) are not working in our CI.

### Steps to reproduce the bug.

I'm not able to provide a simple reproducer (and sorry for that...) One of the tests concerned is `test_01a_sbr_link_array` in https://github.com/ansys/pyaedt/blob/main/_unittest_solvers/test_00_analyze.py 

Here is a relevant output of our [failing CI](https://github.com/ansys/pyaedt/actions/runs/9397216288/job/25879969036)
![image](https://github.com/pyvista/pyvista/assets/146729917/d1175516-13cb-48c2-a453-eda1899b9478)

### System Information

```shell
Succeeding configuration is
--------------------------------------------------------------------------------
  Date: Mon Jun 10 10:03:59 2024 Coordinated Universal Time

                OS : Windows (11 10.0.22631 SP0 Multiprocessor Free)
            CPU(s) : 8
           Machine : AMD64
      Architecture : 64bit
               RAM : 32.0 GiB
       Environment : Python
        GPU Vendor : Mesa/X.org
      GPU Renderer : llvmpipe (LLVM 15.0.6, 256 bits)
       GPU Version : 4.5 (Core Profile) Mesa 22.3.3
  MathText Support : False

  Python 3.12.3 (tags/v3.12.3:f6650f9, Apr  9 2024, 14:05:25) [MSC v.1938 64
  bit (AMD64)]

           pyvista : 0.42.3
               vtk : 9.3.20231014
             numpy : 1.26.4
        matplotlib : 3.8.4
            scooby : 0.10.0
             pooch : 1.8.2
            pillow : 10.3.0
           imageio : 2.34.1
           IPython : 8.25.0
             scipy : 1.13.1
--------------------------------------------------------------------------------

One of the failing ones is 

--------------------------------------------------------------------------------
  Date: Mon Jun 10 13:50:38 2024 Coordinated Universal Time

                OS : Windows (11 10.0.22631 SP0 Multiprocessor Free)
            CPU(s) : 8
           Machine : AMD64
      Architecture : 64bit
               RAM : 32.0 GiB
       Environment : Python
        GPU Vendor : Mesa/X.org
      GPU Renderer : llvmpipe (LLVM 15.0.6, 256 bits)
       GPU Version : 4.5 (Core Profile) Mesa 22.3.3
  MathText Support : False

  Python 3.12.3 (tags/v3.12.3:f6650f9, Apr  9 2024, 14:05:25) [MSC v.1938 64
  bit (AMD64)]

           pyvista : 0.43.9
               vtk : 9.3.0
             numpy : 1.26.4
        matplotlib : 3.8.4
            scooby : 0.10.0
             pooch : 1.8.2
            pillow : 10.3.0
           imageio : 2.34.1
           IPython : 8.25.0
             scipy : 1.13.1
--------------------------------------------------------------------------------
```

**Comment by SMoraisAnsys** [comment]
Just so you know, this problem does not arise when running the test with Python 3.10.9 and `pyvista==0.43.9`

**Comment by tkoyama010** [comment]
Could you please share the loading file which is raise error?

**Comment by SMoraisAnsys** [comment]
> Could you please share the loading file which is raise error? I could not reproduce it by my files.

@tkoyama010 Thanks for looking into this problem. Here is a reproducer that you can use. It requires 2 files ([array_231.aedtz](https://github.com/ansys/pyaedt/blob/main/_unittest_solvers/example_models/T00/array_213.aedtz) and [satellite_231.aedtz](https://github.com/ansys/pyaedt/blob/main/_unittest_solvers/example_models/T00/satellite_231.aedtz))that can be downloaded from https://github.com/ansys/pyaedt/tree/main/_unittest_solvers/example_models/T00. Note that you need to have the file close to the script for it to work. Otherwise, you might end up with project being created instead of using the one that we are using.

Note: when using `vtk-osmesa==9.3.20240525.dev0` I'm able to reproduce the previous stated behavior

![image](https://github.com/pyvista/pyvista/assets/146729917/292b2fb5-4c40-443b-b16b-72c9ac36dc70)

while some other version of `vtk-osmesa` simply lead to a segfault

![image](https://github.com/pyvista/pyvista/assets/146729917/532eb89a-995a-4086-be9e-29739b7489f6)

Here is the python script.
```python
import os

from pyaedt.generic.settings import settings

settings.enable_local_log_file = False
settings.enable_global_log_file = False
settings.number_of_grpc_api_retries = 6
settings.retry_n_times_time_interval = 0.5
settings.enable_error_handler = False
settings.enable_desktop_logs = False
settings.desktop_launch_timeout = 180
settings.release_on_exception = False
settings.disable_bounding_box_sat = True
settings.use_grpc_api = True

from pyaedt import Hfss

SBR_NAME = "satellite_231"
ARRAY_NAME = "array_231"

sbr_platform = Hfss(projectname=r".\satellite_231.aedtz", designname=None, solution_type=None, specified_version="2024.1", non_graphical=False)
array_app = Hfss(projectname=r".\array_231.aedtz", designname=None, solution_type=None, specified_version="2024.1", non_graphical=False)


res = sbr_platform.create_sbr_linked_antenna(array_app, target_cs="antenna_CS", field_type="farfield")
if not res:
    print("Something went wrong when creating SBR linked antenna")

sbr_platform.analyze(cores=4)
ffdata = sbr_platform.get_antenna_ffd_solution_data(frequencies=12e9, sphere="3D")
ffdata2 = sbr_platform.get_antenna_ffd_solution_data(frequencies=12e9, sphere="3D", overwrite=False)

ffdata.plot_2d_cut(quantity="RealizedGain", primary_sweep="theta", secondary_sweep_value=[75], theta=20,
                   title="Azimuth at {}Hz".format(ffdata.frequency), quantity_format="dB10", show=False,
                   image_path="2d1_array.jpg")
if not os.path.exists("2d1_array.jpg"):
    print("Image '2d1_array.jpg' is not found.")

ffdata2.polar_plot_3d_pyvista(quantity="RealizedGain",
                              rotation=[[1, 0.0, 0.0], [0.0, 0.0, 1.0], [0.0, 1.0, 0.0]],
                              image_path="3d2_array.jpg", show=False, convert_to_db=True)
if not os.path.exists("3d2_array.jpg"):
    print("Image '3d2_array.jpg' is not found.")


sbr_platform.close_project(save_project=False)
array_app.close_project(save_project=False)
```

**Comment by tkoyama010** [comment]
@SMoraisAnsys 
Sorry for the late reply. I need your help to make the simple test code. Could you please send us `cad.path` file used in the code below?
```bash
pyaedt\generic\plot.py:1355: in _read_mesh_files
    filedata = pv.read(cad.path)
```

**Comment by SMoraisAnsys** [comment]
> @SMoraisAnsys Sorry for the late reply. I need your help to make the simple test code. Could you please send us `cad.path` file used in the code below?
> 
> ```shell
> pyaedt\generic\plot.py:1355: in _read_mesh_files
>     filedata = pv.read(cad.path)
> ```

Thanks for taking the time to look into this. I'll try to provide you this file tomorrow !

**Comment by SMoraisAnsys** [comment]
Here is a reproducer with `cad.obj` contained in the archive :

```
import pyvista as pv
[cad.zip](https://github.com/user-attachments/files/15952539/cad.zip)

file_data = pv.read("cad.obj")
```

Thanks again for your help on this !

**Comment by tkoyama010** [comment]
Thank you for the data. The error did not occur in the following environment, and I plan to check again by aligning the OS environment with Windows.

I am now guessing that it is most likely an OS dependent issue with `vtk-osmesa`. If so, we will report it to the `vtk` project.
```bash
pip uninstall vtk
pip install vtk-osmesa==9.3.0 --index-url https://gitlab.kitware.com/api/v4/projects/13/packages/pypi/simple
```
```bash
--------------------------------------------------------------------------------
  Date: Mon Jul 01 07:59:42 2024 JST

                OS : Linux (Ubuntu 24.04)
            CPU(s) : 20
           Machine : x86_64
      Architecture : 64bit
       Environment : IPython
        GPU Vendor : Mesa/X.org
      GPU Renderer : llvmpipe (LLVM 15.0.6, 256 bits)
       GPU Version : 4.5 (Core Profile) Mesa 22.3.2
  MathText Support : True

  Python 3.12.3 (main, Apr 10 2024, 05:33:47) [GCC 13.2.0]

           pyvista : 0.43.9
               vtk : 9.3.0
             numpy : 1.26.4
        matplotlib : 3.9.0
            scooby : 0.10.0
             pooch : 1.8.2
            pillow : 10.3.0
           IPython : 8.26.0
--------------------------------------------------------------------------------
```

**Comment by tkoyama010** [comment]
Memo: I found same error in this page (https://docu.ngsolve.org/ngs24/CFD/vtk.html#).

**Comment by tkoyama010** [comment]
> Thanks for the feedback. I can open the issue if you want and forward your feedback on Linux working.

Yes. Please. We welcome contributions.

**Comment by banesullivan** [comment]
@SMoraisAnsys, release 0.44.0 should work for you now to resolve the original issue in your post. I'm pretty sure you were experiencing issues with PyVista's classes' method resolution order and some changes in VTK that caused some properties to be improperly resolved. I fixed this in #5650

**Comment by SMoraisAnsys** [comment]
> @SMoraisAnsys, release 0.44.0 should work for you now to resolve the original issue in your post. I'm pretty sure you were experiencing issues with PyVista's classes' method resolution order and some changes in VTK that caused some properties to be improperly resolved. I fixed this in #5650

I'm going to try it out, thanks for the notification !
